tests/generic_config_updater/test_bgpl.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import pytest
+
+from netaddr import IPNetwork
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.generators import generate_ip_through_default_route
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for bgpmon config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+@pytest.fixture(scope="module")
+def bgpmon_setup_info(duthost):
+    """ Get initial setup info for BGPMONITOR
+    """
+    peer_addr = generate_ip_through_default_route(duthost)
+    pytest_assert(peer_addr, "Failed to generate ip address for test")
+    peer_addr = str(IPNetwork(peer_addr).ip)
+
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    local_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+
+    return peer_addr, local_addr, str(mg_facts['minigraph_bgp_asn'])
+
+def bgpmon_cleanup_config(duthost):
+    """ Clean up BGPMONITOR config to make sure t0 is not broken by other tests
+    """
+    cmds = 'sonic-db-cli CONFIG_DB keys "BGP_MONITORS|*" | xargs -r sonic-db-cli CONFIG_DB del'
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'],
+        "bgpmon cleanup config failed"
+    )
+
+def check_bgpmon_with_addr(duthost, addr):
+    """ Check BGP MONITOR config change is taken into effect
+    """
+    cmds = "show ip bgp summary | grep -w {}".format(addr)
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'],
+        "BGPMonitor with addr {} is not being setup.".format(addr)
+    )
+
+def bgpmon_tc1_add_init(duthost, bgpmon_setup_info):
```

Comment:
>bgpmon_tc1_add_init [](http://example.com/codeflow?start=4&length=19)

Do we need extra testcases to admin up/down an existing peer?

Reply:
Add admin down test for existing bgpmon.